types/src/state_store/state_value.rs

```diff
@@ -31,29 +31,37 @@ use serde::{Deserialize, Serialize};
 )]
 #[cfg_attr(any(test, feature = "fuzzing"), derive(proptest_derive::Arbitrary))]
 pub struct StateValue {
-    pub bytes: Vec<u8>,
+    pub maybe_bytes: Option<Vec<u8>>,
     hash: HashValue,
 }
 
 impl StateValue {
-    fn new(bytes: Vec<u8>) -> Self {
+    fn new(maybe_bytes: Option<Vec<u8>>) -> Self {
         let mut hasher = StateValueHasher::default();
-        hasher.update(&bytes);
+        if let Some(bytes) = &maybe_bytes {
+            hasher.update(bytes);
+        } else {
+            hasher.update(&[]);
```

Comment:
For None we can just use `HashValue::zero()`? Or better, `StateValue` can be `StateValue(Option<StateValueInner>)`, and later when we remove the hack, `StateValueInner` becomes `StateValue`